### DISCUSSION TITLE
Inquiry About Azure OpenAI Support in Cognita

### DISCUSSION THREAD

**Discussion by sandyverden** [truefoundry/cognita#416] [Q&A]
Hello everyone,

I am looking to configure Azure OpenAI as a Model Provider within Cognita, but I haven't found any examples or documentation on this. Can anyone confirm if Cognita supports Azure OpenAI? Any guidance or resources would be greatly appreciated!

Thank you!

**Comment by 2000calories** [comment]
Same problem

**Reply by xuarig007** [reply]
Hi,

Same issue, but you can check the chat/embedding setup with : 

http://localhost:8000/v1/internal/models?model_type=embedding or http://localhost:8000/v1/internal/models?model_type=chat

in my models_config.yaml, i put : 

model_providers:
  - provider_name: azure-openai-gpt4
    api_format: openai
    base_url: https://xxxxxxx-canada.openai.azure.com/
    api_key_env_var: AZURE_OPENAI_API_KEY
    llm_model_ids:
      - gpt-4
    embedding_model_ids: []
    reranking_model_ids: []
    default_headers: {}
    extra:
      api_version: "2023-06-01-preview"
      temperature: 0.7
      max_tokens: 1024
      streaming: true
      system_prompt: "Tu es un assistant intelligent spécialisé en aide technique."

  - provider_name: azure-openai-embeddings
    api_format: openai
    base_url: https://xxxxxxx-canada.openai.azure.com/
    api_key_env_var: AZURE_OPENAI_API_KEY
    llm_model_ids: []
    embedding_model_ids:
      - text-embedding-ada-002
    reranking_model_ids: []
    default_headers: {}
    extra:
      api_version: "2023-06-01-preview"

**Reply by xuarig007** [reply]
I had modified the file backend/modules/model_gateway/model_gateway.py to use AzureChatOpenAI and AzureOpenAIEmbeddings instead of ChatOpenAI and OpenAIEmbeddings

I had modified types.py to add specific parameters for azure

class ModelProviderConfig(ConfiguredBaseModel):
    provider_name: str
    api_format: str
    base_url: Optional[str] = None
    api_key_env_var: str
    default_headers: Dict[str, str] = Field(default_factory=dict)
    llm_model_ids: List[str] = Field(default_factory=list)
    embedding_model_ids: List[str] = Field(default_factory=list)
    reranking_model_ids: List[str] = Field(default_factory=list)
    audio_model_ids: List[str] = Field(default_factory=list)
    -> deployment_name: Optional[str] = None
    -> api_version: Optional[str] = None

The models_config.yaml 


model_providers:
  - provider_name: azure-openai-gpt4
    api_format: openai
    base_url: https://xxxxx-canada.openai.azure.com
    api_key_env_var: AZURE_OPENAI_API_KEY
    llm_model_ids:
      - gpt-4
    embedding_model_ids: []
    reranking_model_ids: []
    default_headers: {}
    api_version: "2023-06-01-preview"
    deployment_name: GPT4
    extra:
      temperature: 0.7
      max_tokens: 1024
      streaming: true
      system_prompt: "Tu es un assistant intelligent spécialisé en aide technique."

  - provider_name: azure-openai-embeddings
    api_format: openai
    base_url: [
    base_url: https://xxxxx-canada.openai.azure.com
    api_key_env_var: AZURE_OPENAI_API_KEY
    llm_model_ids: []
    embedding_model_ids:
      - text-embedding-ada-002
    reranking_model_ids: []
    default_headers: {}
    api_version: "2023-06-01-preview"
    deployment_name: EMBEDDINGS_ADA_002
    extra:
      api_version: "2023-06-01-preview"
      deployment_name: EMBEDDINGS_ADA_002


[model_gateway.zip](https://github.com/user-attachments/files/20706823/model_gateway.zip)

**Reply by xkostrna** [reply]
Did it work ?
